src/sql/src/plan/query.rs

```diff
@@ -3110,6 +3117,30 @@ fn plan_using_constraint(
             }
         }
 
+        // If a `join_using_alias` is present, add a new scope item that accepts
+        // only table-qualified references for each specified join column.
+        // Unlike regular table aliases, a `join_using_alias` should not hide the
+        // names of the joined relations.
+        if let Some(alias_name) = alias {
+            let new_item = PartialItemName {
+                database: None,
+                schema: None,
+                item: alias_name.clone().to_string(),
+            };
+
+            for partial_item_name in both_scope.table_names() {
+                if partial_item_name.matches(&new_item) {
+                    sql_bail!("table name \"{}\" specified more than once", new_item)
+                }
+            }
```

Comment:
Seems odd to do this check inside the loop, for every column name. I think better to hoist it out to L3075, above the loop.

Reply:
Definitely, thanks for catching that!